java/src/main/java/de/gdata/vaas/Vaas.java

```diff
@@ -1,655 +1,544 @@
      */
-    public VaasVerdict forSha256(Sha256 sha256) throws InterruptedException, TimeoutException,
-            VaasInvalidStateException, VaasConnectionClosedException, VaasClientException, VaasServerException {
-        return this.forSha256(sha256, UUID.randomUUID(), null);
+    @Override
+    public CompletableFuture<VaasVerdict> forFileAsync(Path file)
+            throws IOException, VaasAuthenticationException, NoSuchAlgorithmException {
+        return forFileAsync(file, ForFileOptions.fromVaaSConfig(this.config));
     }
 
     /**
-     * Request verdict for Sha256
+     * Asynchronously processes a file and returns a {@link CompletableFuture}
+     * containing the {@link VaasVerdict}.
+     * This method uses default options for file processing, including using the
+     * cache and hash lookup.
      *
-     * @param sha256                   the hashsum to analyze
-     * @param verdictRequestAttributes additional attributes for the request*
-     * @return the Vaas verdict
-     * @throws InterruptedException          if the operation is interrupted by
-     *                                       Thread.interrupt()
-     * @throws TimeoutException              if the request times out
-     * @throws VaasInvalidStateException     if the connection is in an invalid
-     *                                       state
-     * @throws VaasConnectionClosedException if the connection to the Vaas backend
-     *                                       is closed
+     * @param file    the {@link Path} to the file to be processed
+     * @param options The options to customize the request, such as using the cache
+     *                and hash lookup.
+     * @return a {@link CompletableFuture} containing the {@link VaasVerdict} for
+     * the file
+     * @throws IOException                 if an I/O error occurs
+     * @throws VaasAuthenticationException if authentication fails
+     * @throws NoSuchAlgorithmException    if the algorithm for hash lookup is not
+     *                                     available
      */
-    public VaasVerdict forSha256(Sha256 sha256, VerdictRequestAttributes verdictRequestAttributes)
-            throws InterruptedException, TimeoutException,
-            VaasInvalidStateException, VaasConnectionClosedException, VaasClientException, VaasServerException {
-        return this.forSha256(sha256, UUID.randomUUID(), verdictRequestAttributes);
+    @Override
+    public CompletableFuture<VaasVerdict> forFileAsync(Path file, ForFileOptions options)
+            throws IOException, VaasAuthenticationException, NoSuchAlgorithmException {
+        var sha256 = new Sha256(file);
+        var contentLength = Files.size(file);
+        var forSha256Options = new ForSha256Options(options.isUseCache(), options.isUseHashLookup(),
+                options.getVaasRequestId());
+        return forSha256Async(sha256, forSha256Options)
+                .thenCompose(handleException(vaasVerdict -> {
```

Comment:
It is possible that the server has a failed scan in the cache (from a previous failure), which will be returned as an error during forSha256. This then throws the exception here, but it should actually just be ignored: We can just (re-)upload the file and start a new analysis. In general, forSha256 errors should not cause a failure here, but just proceed with the upload & scan (if that also fails, then throw)

Reply:
Which exception dou you mean?

Reply:
Any VaasClientException/VaasServerException that forSha256 might throw